### DISCUSSION TITLE
Disconnect with injected providers in autoConnect mode

### DISCUSSION THREAD

**Discussion by saihaj** [wevm/wagmi#106] [Ideas]
Today in autoConnect mode if you use wallet connect
- Connect account -> refresh: you will stay connected (expected)
- Disconnect account -> refresh: you will be disconnected (expected)

with metamask/injected provider
- Connect account -> refresh: you will stay connected (expected)
- Disconnect account -> refresh: you will be connected (unexpected)

this is for sure a limitation with MM as discussed in 
https://github.com/tmm/wagmi/discussions/68 but one way I can think we can create a same behavior as walletConnect is by making a few changes to how it is connected on mount https://github.com/tmm/wagmi/blob/0336e9aa7a2acdd4a75214fe0be551bd47f42e77/packages/react/src/context.ts#L110 would need to store an additional variable in local storage and check it to avoid autoConnect on page refreshes for injectedProviders.

edit: if this is of interest I can send in a PR cause this makes the functionality more consistent.

**Comment by tmm** [comment]
What do you think about adding a `shimDisconnect` option to `InjectedConnector`? When this option is switched on, connection status is handled in local storage a la WalletConnect.

There might be some edge cases around disconnecting directly through the MM UI that we'll want to make sure are handled.

**Reply by saihaj** [reply]
So this would be an option to provider to tell use `shimDisconect`? Or when user uses `autoConnect` just default to `shimDisconnect`?

**Reply by tmm** [reply]
Seems good to make this an option of the connector directly:

```ts
import { InjectedConnector } from 'wagmi/connectors/injected'

const connector = new InjectedConnector({
  options: {
    shimDisconnect: true,
  },
})
```

Then, check for the `shimDisconnect` flag in `connect`, `disconnect`, `isAuthorized`, etc. to modify the behavior.

Happy to review and give PR feedback if you want to take a first pass :)

**Reply by saihaj** [reply]
Sounds good will try to work on something

**Reply by tmm** [reply]
Added https://github.com/tmm/wagmi/pull/119

**Reply by tmm** [reply]
Update: `shimDisconnect` will be the default behavior for `InjectedConnector` in `0.3.0`

**Reply by dwasyluk** [reply]
Can you provide any more info on how to use `shimDisconnect`? I have it enabled but I don't quite understand how to trigger the user re-selecting their account after disconnecting when using Metamask. It seems like even with `shimDisconnect` the wagmi libs don't prompt the user to re-select their account when calling `connect()` from a disconnected state. Its possible to do this using the raw ethers methods but I'm not sure how to trigger an equivalent experience using walletconnect/wagmi and `shimDisconnect`?
